dashboard/src/app/stacks/stack-details/import-stack.service.ts

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) 2015-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ */
+'use strict';
+
+interface IValidation {
+  isValid: boolean;
+  errors: Array<string>;
+}
+
+const COMPOSE_FILE = 'compose';
+const DOCKERFILE = 'dockerfile';
+
+/**
+ * This class is handling the data for import stack and it's validator
+ *
+ * @author Oleksii Orel
+ */
+export class ImportStackService {
+  stack: che.IStack | {};
+
+
+  /**
+   * Default constructor that is using resource
+   * @ngInject for Dependency injection
+   */
+  constructor() {
+    this.stack = null;
+  }
+
+  /**
+   * Sets imported stack.
+   *
+   * @param stack {che.IStack}
+   */
+  setStack(stack: che.IStack | {}): void {
+    this.stack = stack;
+  }
+
+  /**
+   * Returns imported stack.
+   *
+   * @returns {che.IStack}
+   */
+  getStack(): che.IStack | {} {
+    return this.stack;
+  }
+
+  /**
+   * Returns valid status of the stack.
+   *
+   * @returns {boolean}
+   */
+  isValid(): boolean {
+    return this.getStackValidation(this.stack).isValid;
+  }
+
+  /**
+   * Return result of recipe validation.
+   * @param stack {che.IStack}
+   * @returns {IValidation}
+   */
+  getStackValidation(stack: che.IStack | {}): IValidation {
```

Comment:
Should this method be in separate service? I think it could be used not only for stack import.